### PR TITLE
A/B flag for remote write

### PR DESCRIPTION
**Pull request by brancz**
This implements the remote write feature, but turns it off by default, but at the same time allows using the new, much simpler configuration mechanism for selecting metrics while using the more conservative v1 protocol.

@LiliC @s-urbaniak

### CODE REVIEW

**Comment by brancz** [comment]
/hold

Just to check in CI that indeed the v1 protocol is used.

**Comment by brancz** [comment]
Validated that this works as expected.

/hold cancel

File: cmd/operator/main.go [hunk 1]
```diff
@@ -86,10 +91,35 @@ func Main() int {
 	kubeconfigPath := flagset.String("kubeconfig", "", "The path to the kubeconfig to connect to the apiserver with.")
 	apiserver := flagset.String("apiserver", "", "The address of the apiserver to talk to.")
 	releaseVersion := flagset.String("release-version", "", "Currently targeted release version to be reconciled against.")
+	telemetryConfigFile := flagset.String("telemetry-config", "/etc/cluster-monitoring-operator/telemetry/metrics.yaml", "Path to telemetry-config.")
+	remoteWrite := flagset.Bool("enabled-remote-write", false, "Wether to use legacy telemetry write protocol or Prometheus remote write.")
```
Comment (metalmatze) [inline cmd/operator/main.go]: So this is the real change now, is it?

**Comment by LiliC** [comment]
@brancz don't forget to update the docs :)

File: cmd/operator/main.go (hunk 1 above)
Comment (brancz) [reply cmd/operator/main.go]: Yep

**Pull request merged**